lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -50,9 +50,20 @@ sub _allocate_jobs ($self, $free_workers) {
     my $max_allocate = $limit >= 0 ? min(MAX_JOB_ALLOCATION, $limit - $running) : MAX_JOB_ALLOCATION;
 
     # update the matching workers to the current free
-    for my $jobinfo (values %$scheduled_jobs) {
-        $jobinfo->{matching_workers} = _matching_workers($jobinfo, $free_workers);
+    my %rejected;
+    for my $id (keys %$scheduled_jobs) {
+        my $jobinfo = $scheduled_jobs->{$id};
+        $jobinfo->{matching_workers} = _matching_workers($jobinfo, $free_workers, \%rejected);
+        delete $scheduled_jobs->{$id} unless @{$jobinfo->{matching_workers}};
     }
+    if (keys %rejected) {
+        my @rejected = sort { $rejected{$b} <=> $rejected{$a} || $a cmp $b } keys %rejected;
+        @rejected = @rejected[0 .. 29] if @rejected > 30;
+        my $stats = join ',', map { "$_:$rejected{$_}" } @rejected;
+        my $info = sprintf "Skipping %d jobs because of missing workers (%s)", sum(values %rejected), $stats;
```

Comment:
what does "missing" mean in this context?

Reply:
How about this:
```suggestion
        my $info = sprintf "Skipping %d jobs because of no free workers with requested worker classes (%s)", sum(values %rejected), $stats;
```